### DISCUSSION TITLE
Date Input Group

### DISCUSSION THREAD

**Discussion by Rozerinay** [nl-design-system/backlog#271] [Component suggestions]
<img width="480" alt="visuele weergave van het date input group component" src="https://github.com/nl-design-system/backlog/assets/84851545/f7c8e706-6548-4e1f-a5c2-8b27d935e955">

We willen graag meer leren over het Date Input Group component. In welke vorm, of vormen, wordt deze momenteel ingezet en toegepast binnen jouw organisatie?

Deel hieronder een link naar Figma, Storybook, Github of screenshots. Met deze informatie kunnen we bewijzen dat het component onderdeel moet worden van de NL Design System component library.

Daarnaast wordt het delen van onderzoek gewaardeerd. Dus heb je zelf onderzoek gedaan en zijn er inzichten voor dit component? Dan horen we dit ook graag!

[Link naar bericht in Slack](https://codefornl.slack.com/archives/C7KDQRXMZ/p1718791870878859)

---

## Naam
Date Input Group

## Doel
Formulierveld voor een datum die de gebruiker al weet of die van bekende gegevens overgenomen kan worden.

## Ook bekend als
Date Input

## Gerelateerde componenten
[Calendar](https://github.com/orgs/nl-design-system/discussions/273), [Date Input](https://github.com/orgs/nl-design-system/discussions/270), [Date Picker](https://github.com/orgs/nl-design-system/discussions/272), [Input Group](https://github.com/orgs/nl-design-system/discussions/296)

## Onderzoek

[Date Input component - GOV.UK](https://design-system.service.gov.uk/components/date-input/)
Date Input component zoals beschikbaar bij GOV.UK.

[Date Input Github Discussion - GOV.UK](https://github.com/alphagov/govuk-design-system-backlog/issues/42)
In deze Discussion van GOV.UK wordt het Date Input component besproken. Er wordt onder andere [vermeld](https://github.com/alphagov/govuk-design-system-backlog/issues/42#issuecomment-1119724868) dat er gebruikers zijn die de naam van de maand (janunari of JAN) invullen in plaats van het nummer (1). Er wordt ook [geschreven](https://github.com/alphagov/govuk-design-system-backlog/issues/42#issuecomment-1398208356) waarom er niet voor 1 veld (eventueel met 'mask'), en auto-tab is gekozen.

[Why we care more about effectiveness than efficiency of satisfaction - Blog - GOV.UK](https://userresearch.blog.gov.uk/2017/04/18/why-we-care-more-about-effectiveness-than-efficiency-or-satisfaction/)
In deze blogpost wordt er uitgelegd waarom GOV.UK expres niet kiest voor het inzetten van auto-tab.

[Designing Birthday Picker UX: Simpler Is Better - Smashing magazine](https://www.smashingmagazine.com/2021/05/frustrating-design-patterns-birthday-picker/)
Artikel waarin wordt beschreven waarom Date Input en Date Pickers voor frustratie kunnen zorgen voor een geboortedatum.

[Format the “Expiration Date” Fields Exactly the Same as the Physical Credit Card - Baymard](https://baymard.com/blog/how-to-format-expiration-date-fields)
Artikel waarin wordt beschreven wat de beste manier is om de vervaldatum van een creditcard uit te vragen.

**Comment by jeffreylauwers** [comment]
## DUO

[Documentatie website](https://uno.dfront.rijkscloud.nl/#/components/date-input)

---

### Screenshots / Varianten

**Documentatie website**

Default
![Scherm­afbeelding 2024-12-04 om 14 48 26](https://github.com/user-attachments/assets/92c9e95c-a266-4ba5-8094-6d3e1a05159a)

Filled + Focus
![Scherm­afbeelding 2024-12-04 om 15 01 38](https://github.com/user-attachments/assets/b16d4709-ce48-48e6-a923-02be7e6e2e7f)

Invalid
![Scherm­afbeelding 2024-12-04 om 14 48 32](https://github.com/user-attachments/assets/981cb83a-ca2e-4c9a-9e85-7408234a36b7)

---

### Notities

- Binnen een `fieldset` met `legend`.
- CSS class van groep invulvelden is `input__control-group`.
- CSS class van enkel invulveld binnen deze groep is `input__control-container`.
- Invulvelden voor de data zijn op basis van `type="number"`.
- Opvallend is dat 'Dag' `min="1"`, `max="31"` en `[maxlength="2"](https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/maxlength)` als toevoeging heeft, maar dat deze niet lijken te werken?
- Opvallend is dat in Firefox (Mac) er stepper controls binnen het invulveld bij komen omdat deze van het type 'number' zijn. Deze nemen dusdanig veel ruimte in dat de ingevoerde data niet goed zichtbaar is. Dit is wel te [verhelpen](https://stackoverflow.com/questions/3790935/can-i-hide-the-html5-number-input-s-spin-box).

![Screenshot van Date Input component in Firefox](https://github.com/user-attachments/assets/a0a3d06a-49d3-453c-b43c-5d589841ee0b)

**Comment by jeffreylauwers** [comment]
## Open Formulieren

[Storybook](https://open-formulieren.github.io/open-forms-sdk/?path=/docs/pure-react-components-forms-datefield-input-group--docs)

---

### Screenshots / Varianten

**Storybook**

Default
![Scherm­afbeelding 2024-12-04 om 15 01 08](https://github.com/user-attachments/assets/fd3bf9db-a66a-4acf-825a-1702f3070623)

Filled + Focus
![Scherm­afbeelding 2024-12-04 om 15 02 52](https://github.com/user-attachments/assets/778d13fc-0214-4825-b29b-8892d149e2a4)

Invalid
![Scherm­afbeelding 2024-12-04 om 15 04 02](https://github.com/user-attachments/assets/a50745c3-54a2-4114-8ebf-9670e3723e45)

Value normalizes to ISO-8601
![Scherm­afbeelding 2024-12-04 om 15 03 46](https://github.com/user-attachments/assets/88946748-5037-4e12-a470-6d39bb8b935f)

---

### Notities

- Binnen een `fieldset` met `legend` (met daarbinnen weer een `label`?).
- CSS class van groep invulvelden is `openforms-input-group`.
- CSS class van enkel invulveld binnen deze groep is `openforms-input-group__item`.
- Invulvelden zelf zijn op basis van `type="text"` met `inputmode="numeric".
- Geen `maxlength` toevoeging.
- Er is een [ISO 8601](https://en.wikipedia.org/wiki/ISO_8601) story beschikbaar.

**Comment by Rubenoo** [comment]
Wij gebruiken dit bij https://rni-webform.rvig.nl/nl
![image](https://github.com/user-attachments/assets/f6be26d9-1cd3-425a-bf88-36321f1cfc4f)

Bij het openen van de date input wordt automatisch de standaard datumkiezer van de browser gebruikt.

In het verleden hebben we veel tijd geïnvesteerd in het ontwikkelen van een aangepaste datuminvoer op basis van de MUI datepicker, waarbij de maand voluit geschreven moest worden. Dit deden we om verwarring te voorkomen tussen verschillende datumnotaties (DDMMJJJJ en MMDDJJJJ). 
Echter, de technische implementatie voor alle ondersteunde talen (15 in totaal) bleek zeer complex. Bovendien bleek uit gebruikersonderzoek dat deze oplossing niet gebruiksvriendelijk was. Daarom zijn we teruggekeerd naar de standaard datuminvoer (input type=date).

**Reply by jeffreylauwers** [reply]
Bedankt voor je bijdrage! Deze vorm komt overeen met de Date Input component, daar zijn we de [Date Input - Github Discussion](https://github.com/orgs/nl-design-system/discussions/270) voor gestart.

**Comment by jeffreylauwers** [comment]
## ✨ Dit component is nu Help Wanted ✨

Help je mee hem door de Community stappen te krijgen?
[Date Input Group op het Community bord](https://github.com/orgs/nl-design-system/projects/29/views/1?filterQuery=date+input+group)
